aws/resource_aws_wafv2_regex_pattern_set.go

```diff
@@ -0,0 +1,268 @@
+		},
+	}
+}
+
+func resourceAwsWafv2RegexPatternSetCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).wafv2conn
+	params := &wafv2.CreateRegexPatternSetInput{
+		Name:                  aws.String(d.Get("name").(string)),
+		Scope:                 aws.String(d.Get("scope").(string)),
+		RegularExpressionList: []*wafv2.Regex{},
+	}
+
+	if v, ok := d.GetOk("description"); ok {
+		params.Description = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("regular_expression_list"); ok && v.(*schema.Set).Len() > 0 {
+		params.RegularExpressionList = expandWafv2RegexPatternSet(v.(*schema.Set).List())
+	}
+
+	if v := d.Get("tags").(map[string]interface{}); len(v) > 0 {
+		params.Tags = keyvaluetags.New(v).IgnoreAws().Wafv2Tags()
+	}
+
+	resp, err := conn.CreateRegexPatternSet(params)
+
+	if err != nil {
+		return fmt.Errorf("Error creating WAFv2 RegexPatternSet: %s", err)
+	}
+
+	d.SetId(*resp.Summary.Id)
+
+	return resourceAwsWafv2RegexPatternSetRead(d, meta)
+}
+
+func resourceAwsWafv2RegexPatternSetRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).wafv2conn
+	ignoreTagsConfig := meta.(*AWSClient).IgnoreTagsConfig
+	params := &wafv2.GetRegexPatternSetInput{
+		Id:    aws.String(d.Id()),
+		Name:  aws.String(d.Get("name").(string)),
+		Scope: aws.String(d.Get("scope").(string)),
+	}
+
+	resp, err := conn.GetRegexPatternSet(params)
+	if err != nil {
+		if isAWSErr(err, wafv2.ErrCodeWAFNonexistentItemException, "AWS WAF couldn’t perform the operation because your resource doesn’t exist") {
+			log.Printf("[WARN] WAFv2 RegexPatternSet (%s) not found, removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+
+		return err
+	}
+
+	d.Set("name", resp.RegexPatternSet.Name)
+	d.Set("description", resp.RegexPatternSet.Description)
+	d.Set("arn", resp.RegexPatternSet.ARN)
+	d.Set("lock_token", resp.LockToken)
+
+	if err := d.Set("regular_expression_list", flattenWafv2RegexPatternSet(resp.RegexPatternSet.RegularExpressionList)); err != nil {
+		return fmt.Errorf("Error setting regular_expression_list: %s", err)
+	}
+
+	tags, err := keyvaluetags.Wafv2ListTags(conn, *resp.RegexPatternSet.ARN)
+	if err != nil {
+		return fmt.Errorf("Error listing tags for WAFv2 RegexPatternSet (%s): %s", *resp.RegexPatternSet.ARN, err)
+	}
+
+	if err := d.Set("tags", tags.IgnoreAws().IgnoreConfig(ignoreTagsConfig).Map()); err != nil {
+		return fmt.Errorf("Error setting tags: %s", err)
+	}
+
+	return nil
+}
+
+func resourceAwsWafv2RegexPatternSetUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).wafv2conn
+
+	log.Printf("[INFO] Updating WAFv2 RegexPatternSet %s", d.Id())
+
+	params := &wafv2.UpdateRegexPatternSetInput{
+		Id:                    aws.String(d.Id()),
+		Name:                  aws.String(d.Get("name").(string)),
+		Scope:                 aws.String(d.Get("scope").(string)),
+		Description:           aws.String(d.Get("description").(string)),
```

Comment:
since L177 takes care of conditionally setting the the `Description` key below, I think it's safe to omit it here (docs also suggest the `Description` attribute isn't required)